### PR TITLE
Dart code generator beginnings

### PR DESCRIPTION
**Pull request by JonasWanke**
* <kbd>Core</kbd>: Add `Process.run(…)`, `ProcessResult`
* <kbd>CodeGeneratorDart</kbd>: generate `pubspec.yaml` and execute `dart pub get`

### CODE REVIEW

File: packages/code_generator_dart/src/compile.candy [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    let output = MutableMap.empty<Path, String>()
+
+    let pubspec = generatePubspec(context, package)
+    output.set(pubspec.first, pubspec.second)
+
+    // TODO(JonasWanke): generate code
+
+    Tuple(output, List.empty<CompilerError>())
+  })
+}
```
Comment (MarcelGarus) [inline packages/code_generator_dart/src/compile.candy:43]: Maybe make this a method on `Package`? Usable as `let files = somePackage.compile(context)`.

File: packages/code_generator_dart/src/compile.candy [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  sdk: '>=2.12.0-0 <3.0.0'
+dependencies:
+  characters: ^1.1.0
+  meta: ^1.3.0
+dev_dependencies:
+  test: ^1.16.3"
+
+    Tuple(Tuple(path, content), List.empty<CompilerError>())
+  })
+}
```
Comment (MarcelGarus) [inline packages/code_generator_dart/src/compile.candy:60]: Same here: `myPackage.generatePubspec(context)` would be cleaner.

**Review by MarcelGarus — APPROVED** [review]

File: packages/code_generator_dart/src/compile.candy (hunk 1 above)
Comment (JonasWanke) [reply packages/code_generator_dart/src/compile.candy:43]: We'll have to refactor this when our compiler supports impls for types from other packages.

File: packages/code_generator_dart/src/compile.candy (hunk 2 above)
Comment (JonasWanke) [reply packages/code_generator_dart/src/compile.candy:60]: See the other comment.

**Pull request merged**